components/nexus-repository/src/main/java/org/sonatype/nexus/repository/partial/PartialFetchHandler.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.partial;
+
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.sonatype.nexus.repository.http.HttpResponses;
+import org.sonatype.nexus.repository.view.Context;
+import org.sonatype.nexus.repository.view.Handler;
+import org.sonatype.nexus.repository.view.Headers;
+import org.sonatype.nexus.repository.view.Payload;
+import org.sonatype.nexus.repository.view.PayloadResponse;
+import org.sonatype.nexus.repository.view.Request;
+import org.sonatype.nexus.repository.view.Response;
+import org.sonatype.nexus.repository.view.Status;
+
+import com.google.common.collect.Range;
+import org.apache.http.HttpStatus;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Implements partial-fetch semantics (as per RFC 2616) for {@link Status#isSuccessful() successful} {@link
+ * PayloadResponse}s.
+ *
+ * @since 3.0
+ */
+@Named
+@Singleton
+public class PartialFetchHandler
+    implements Handler
+{
+  private final RangeParser rangeParser;
+
+  @Inject
+  public PartialFetchHandler(final RangeParser rangeParser) {
+    this.rangeParser = checkNotNull(rangeParser);
+  }
+
+  @Nonnull
+  @Override
+  public Response handle(@Nonnull final Context context) throws Exception {
+    final Response response = context.proceed();
+
```

Comment:
Minor but: as per RFC, Range header makes sense only with GET method. Handler might check if action is `GET` and choose to not interfere with any other action.